pkg/stanza/fileconsumer/scanner.go

```diff
@@ -11,27 +11,25 @@ import (
 	stanzaerrors "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/errors"
 )
 
-const defaultBufSize = 16 * 1024
-
 // PositionalScanner is a scanner that maintains position
 type PositionalScanner struct {
 	pos int64
 	*bufio.Scanner
 }
 
 // NewPositionalScanner creates a new positional scanner
-func NewPositionalScanner(r io.Reader, maxLogSize int, startOffset int64, splitFunc bufio.SplitFunc) *PositionalScanner {
+func NewPositionalScanner(r io.Reader, maxLogSize int, bufferCap int, startOffset int64, splitFunc bufio.SplitFunc) *PositionalScanner {
 	ps := &PositionalScanner{
 		pos:     startOffset,
 		Scanner: bufio.NewScanner(r),
 	}
 
-	buf := make([]byte, 0, defaultBufSize)
+	buf := make([]byte, 0, bufferCap)
 	ps.Scanner.Buffer(buf, maxLogSize*2)
 
 	scanFunc := func(data []byte, atEOF bool) (advance int, token []byte, err error) {
 		advance, token, err = splitFunc(data, atEOF)
-		if (advance == 0 && token == nil && err == nil) && len(data) >= 2*maxLogSize {
+		if (advance == 0 && token == nil && err == nil) && len(data) >= maxLogSize {
```

Comment:
This is a related bug which I can figure out a way to trigger when also resolving #22936

Basically, I found some rare cases (specifically noticed in `tiny_max_log_size` test above) where `len(data)` falls between `maxLogSize` and `2*maxLogSize`, so doesn't trigger the condition. At the same time, `token` is empty because data was read into the buffer no token was completed by the split func. The result is that the log was not truncated properly.